--- a/viewer/lib/worker.js	2026-07-25 00:32:38.549512894 +0800
+++ b/viewer/lib/worker.js	2026-07-25 00:32:38.549810252 +0800
@@ -32,7 +32,7 @@
   if (!value) {
     delete dirtySections[key]
     postMessage({ type: 'sectionFinished', key })
-  } else if (chunk && chunk.sections[Math.floor(y / 16)]) {
+  } else if (chunk && chunk.sections[Math.floor((y - (chunk.minY ?? 0)) / 16)]) {
     dirtySections[key] = value
   } else {
     postMessage({ type: 'sectionFinished', key })
@@ -74,7 +74,7 @@
     y = parseInt(y, 10)
     z = parseInt(z, 10)
     const chunk = world.getColumn(x, z)
-    if (chunk && chunk.sections[Math.floor(y / 16)]) {
+    if (chunk && chunk.sections[Math.floor((y - (chunk.minY ?? 0)) / 16)]) {
       delete dirtySections[key]
       const geometry = getSectionGeometry(x, y, z, world, blocksStates)
       const transferable = [geometry.positions.buffer, geometry.normals.buffer, geometry.colors.buffer, geometry.uvs.buffer]
